### PR TITLE
Fix managed channel is not closed

### PR DESCRIPTION
**Pull request by zhexuany**
channelFactory is not closed after tisession being closed. 

This fixes https://github.com/pingcap/tispark/issues/621

### CODE REVIEW

File: tikv-client/src/main/java/com/pingcap/tikv/TiSession.java [hunk 1]
```diff
@@ -205,10 +173,10 @@ public void close() throws Exception {
     getThreadPoolForTableScan().shutdownNow();
     getThreadPoolForIndexScan().shutdownNow();
     getPDClient().close();
+    channelFactory.close();
```
Comment (birdstorm) [inline tikv-client/src/main/java/com/pingcap/tikv/TiSession.java]: I think ChannelFactory should be closed before PDClient.

File: tikv-client/src/main/java/com/pingcap/tikv/util/ChannelFactory.java [hunk 2]
```diff
@@ -52,9 +52,14 @@ public ManagedChannel getChannel(String addressStr) {
...
   @Override
   public void close() {
-    for (ManagedChannel ch : connPool.values()) {
-      ch.shutdown();
-    }
+    connPool.forEach(
+        (k, v) -> {
+          v.shutdownNow();
+          try {
+            v.awaitTermination(1, TimeUnit.SECONDS);
```
Comment (marsishandsome) [inline tikv-client/src/main/java/com/pingcap/tikv/util/ChannelFactory.java]: why wait for 1 second?

File: tikv-client/src/main/java/com/pingcap/tikv/TiSession.java (hunk 1 above)
Comment (zhexuany) [reply tikv-client/src/main/java/com/pingcap/tikv/TiSession.java]: agreed.

File: tikv-client/src/main/java/com/pingcap/tikv/util/ChannelFactory.java (hunk 2 above)
Comment (zhexuany) [reply tikv-client/src/main/java/com/pingcap/tikv/util/ChannelFactory.java]: No particular reason. This is added to make a graceful shutdown.

**Pull request merged**